### PR TITLE
chore(deps): bump @testing-library/user-event from 13.5.0 to 14.6.1

### PR DESCRIPTION
**Pull request by frontend-specialisten**
## Summary

- Bump `@testing-library/user-event` from `13.5.0` to `14.6.1` (last major version, also consolidates with the version Storybook already required transitively).
- Migrate 17 spec files from the v13 sync API to the v14 async pattern: each test now calls `userEvent.setup()` and awaits every interaction.
- Convert the disabled-button assertion in `watt-button.component.spec.ts` from `expect(() => userEvent.click(...)).toThrow()` to `await expect(user.click(...)).rejects.toThrow()` so it still verifies the click is rejected.
- Replace the deprecated per-call `skipPointerEventsCheck: true` with the global `pointerEventsCheck: 0` option in the few specs that need to click disabled elements (timepicker, checkbox, segmented-buttons).
- Work around a happy-dom + Material edge case in the modal `closes on ESC` spec: user-event v14 no longer populates the deprecated `keyCode` property, but `MatDialog` filters Escape by `keyCode === 27`. The test now dispatches a synthetic `KeyboardEvent` with `keyCode` defined explicitly. The companion `disables ESC` test is unaffected because it asserts the modal stays open regardless of the event source.

Ref: #2481

### CODE REVIEW

**Review by mimse — DISMISSED** [review]
🥇

**Review by dzhavat — DISMISSED** [review]

**Review by mimse — CHANGES_REQUESTED** [review]

File: libs/watt/package/drawer/watt-drawer.component.spec.ts [hunk 1]
```diff
@@ -275,8 +286,9 @@ describe(WattDrawerComponent, () => {
 
   it('does not call "closed" on Escape when drawer is closed', async () => {
     await setup(defaultTemplate);
+    const user = userEvent.setup();
 
-    userEvent.keyboard('{Escape}');
+    await user.keyboard('{Escape}');
```
[the remark this answers is not in the document]
Comment (frontend-specialisten) [reply libs/watt/package/drawer/watt-drawer.component.spec.ts]: You are right, and this one I should have caught when you flagged the modal one. The same false-positive pattern, different mechanism: `user.keyboard('{Escape}')` sends the event to the focused element (body when the drawer never opened), and the keydown then bubbles upward and never reaches the `watt-drawer` host listener. So the test passed regardless of whether `close()` actually short-circuits on `!isOpen`.

Fixed in da1564520a by dispatching the synthetic keydown directly on the drawer host element, mirroring the existing `closes on Escape` tests in the same file. Those existing tests use the identical pattern and pass only because the dispatch reaches the listener and triggers `close()` on an open drawer, so we now know the listener is reached. The new assertion that `closed` is not called is a real verification of the closed-drawer branch.

**Review by mimse — APPROVED** [review]
